core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -537,6 +537,14 @@ trait WhiskActionsApi extends WhiskCollectionAPI with PostActionActivation with
 
     val exec = content.exec getOrElse action.exec
 
+    var newAnnotations = action.annotations
```

Comment:
I think we would not want to use the `var` variable.

Since the main target here is to remove items from `action.annotations` which are in the `content.delAnnotations`, I think we can do something similar to this:

```suggestion
    val newAnnotations = content.delAnnotations.map { annotationArray =>
      annotationArray.foldRight(action.annotations)((a: String, b: Parameters) => b - a)
    }.map( _ ++ content.annotations).getOrElse(content.annotations ++ action.annotations)
```

Reply:
Updated accordingly